pagerduty/resource_pagerduty_team_membership.go

```diff
@@ -73,7 +80,21 @@ func resourcePagerDutyTeamMembershipRead(d *schema.ResourceData, meta interface{
 		return err
 	}
 
-	if !isTeamMember(user, teamID) {
+	d.Set("role", "")
+
+	if isTeamMember(user, teamID) {
+		resp, _, err := client.Teams.GetMembers(teamID, &pagerduty.GetMembersOptions{})
```

Comment:
Shouldn't `client.Teams.GetMembers` become the main API to query for the `resourcePagerDutyTeamMembershipRead` instead of calling `client.Users.Get`?

This could probably simplify things a bit, isn't it?

Reply:
Instead of using `client.Users.Get`, I could simplify the logic with `client.Teams.GetMembers` only.
https://github.com/terraform-providers/terraform-provider-pagerduty/pull/151/commits/1f71c0c05f76189fbf45cd27215d08097a81740b

How about this?